modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java

```diff
@@ -90,11 +95,40 @@ public static void handleInbound(CarbonMessage cMsg, CarbonCallback callback) {
 
             // Find the Resource
             ResourceInfo resource = resourceDispatcher.findResource(service, cMsg, callback);
-
-            invokeResource(cMsg, callback, resource, service);
-
+            // engage Service interceptors.
+            CarbonCallback resourceCallback = callback;
+            if (serviceInterceptors.isEnabled(protocol)) {
```

Comment:
Yes, we can make this to static method, which returns Carboncallback, but this requires additional null check. This callback can be null either result.getMessageIntercepted() == null or !result.isInvokeNext().

If callback is null then we should return from the function.